packages/server/lib/browsers/cdp_automation.ts

```diff
@@ -269,6 +269,8 @@ export class CdpAutomation {
         return true
       case 'remote:debugger:protocol':
         return this.sendDebuggerCommandFn(data.command, data.params)
+      case 'remote:debugger:event':
+        return this.onFn(data.event, data.callback)
```

Comment:
I understand why you want to add this. For additional context, we are planning to add official CDP support through a nicer API in the future, see [this comment](https://github.com/cypress-io/cypress/issues/7942#issuecomment-856140184) for some ideas.

One part that is TBD is how the planned API will handle selecting CDP `Target`s. To make your change forward-compatible with the future feature, can you add a `targetId` to the callback to indicate that this event is originating from Cypress's main CDP `Target`? Like, instead of calling `data.callback` with the raw `eventData`, call it with `{ targetId: 'main', eventData }`, or something like that.

That way we can add CDP `Target` selection in a future PR that is forward-compatible with this change (for #7942). For the Cypress-owned `Target`, `targetId` could be `main`, for all others it could be the `targetId` returned by `Target.createTarget`.